nats/aio/msg.py

```diff
@@ -113,29 +106,31 @@ async def nak(self) -> None:
         nak negatively acknowledges a message delivered by JetStream triggering a redelivery.
         """
         self._check_reply()
-        await self._client.publish(self.reply, Msg.Ack.Nak)
+        await self.client.publish(self.reply, Msg.Ack.Nak)
         self._ackd = True
 
-    async def in_progress(self):
+    async def in_progress(self) -> None:
         """
         in_progress acknowledges a message delivered by JetStream is still being worked on.
         Unlike other types of acks, an in-progress ack (+WPI) can be done multiple times.
         """
         if self.reply is None or self.reply == '':
             raise NotJSMessageError
-        await self._client.publish(self.reply, Msg.Ack.Progress)
+        await self.client.publish(self.reply, Msg.Ack.Progress)
 
     async def term(self) -> None:
         """
         term terminates a message delivered by JetStream and disables redeliveries.
         """
         self._check_reply()
 
-        await self._client.publish(self.reply, Msg.Ack.Term)
+        await self.client.publish(self.reply, Msg.Ack.Term)
         self._ackd = True
 
+    # TODO(@orsinium): use a cached_property. Available in functools since 3.8,
+    # as a package (backports.cached-property), or can be just copy-pasted in the project.
```

Comment:
sounds good to constrain to 3.8 since that is the version where `loop` parameter got deprecated.

Reply:
I don't want to bring so radical changes just yet. Let's break things in a separate PR 😉